docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,204 @@
+
+Examples are classified as *zero-shot learning* or *few-shot learning* based on whether they contain verbatim completions.
+
+- **Zero-shot learning** examples include a prompt with no verbatim completion. Because they don't include verbatim completions, zero-shot prompts test a model's responses without giving it example data output. (Zero-shot prompts can have  completions that include cues, such as indicating the model should output an ordered list by including **"1."** as the completion.)
+- **Few-shot learning** examples include several pairs of prompts with verbatim completions. Few-shot learning can change the model's behavior by adding to its existing knowledge.
+
+## Cues
+
+A cue is text that conveys the desired structure or format of output. Like an instruction, a cue isn't processed by the model as if it were user input. Like an example, a cue shows the model what you want instead of telling it what to do. You can add as many cues as you want, so you can iterate to get the result you want. Cues are used with an instruction or an example and should be at the end of the prompt.
+
+Suppose you use an instruction to tell the model to produce a list of presidential accomplishments by category, along with supporting content that tells the model what categories to use. You decide that you want the model to produce a nested list with all caps for categories, with each president's accomplishments in each category listed on one line that begins with their name, with presidents listed chronologically. After your instruction and supporting content, you could add three cues to show the model how to structure and format the list:
+
+```csharp
+prompt = """
+Instructions: Summarize US Presidential accomplishments, grouped by category.
+Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration.
+Accomplishments: George Washington
+First president of the United States.
+First president to have been a military veteran.
+First president to be elected to a second term in office.
+First president to receive votes from every presidential elector in an election.
+First president to fill the entire body of the United States federal judges; including the Supreme Court.
+First president to be declared an honorary citizen of a foreign country, and an honorary citizen of France.
+John Adams ...  /// Text truncated
+
+DOMESTIC POLICY
+- George Washington: 
+- John Adams:
+""";
+```
+
+- **DOMESTIC POLICY** shows the model that you want it to start each group with the category in all caps.
+- **- George Washington:** shows the model to start each section with George Washington's accomplishments listed on one line.
+- **- John Adams:** shows the model that it should list remaining presidents in chronological order.
+
+## .NET implementations
+
+This section compares two .NET options for prompt engineering: [Semantic Kernel](/semantic-kernel/overview/) and [Azure OpenAI](/azure/ai-services/openai/overview).
```

Comment:
I don't think Semantic Kernel and Azure OpenAI should link to other pages here, if anything, I'd expect these to link to the sections below. But then you can add those links off to other pages in the sections themselves.

Reply:
Thanks! revised